test/credentials.spec.coffee

```diff
@@ -370,6 +491,84 @@ if AWS.util.isNode()
         expect(creds.expireTime).to.eql(new Date(0))
         done()
 
+    it 'will assume a role from the credentials file whose source profile is defined in the config file', (done) ->
+      process.env.AWS_SDK_LOAD_CONFIG = '1'
+      helpers.spyOn(AWS.util, 'readFileSync').andCallFake (path) ->
+        if (path.match(/[\/\\]home[\/\\]user[\/\\].aws[\/\\]credentials/))
+          '''
+          [default]
+          aws_access_key_id = akid
+          aws_secret_access_key = secret
+          role_arn = arn
+          source_profile = foo
+          '''
+        else
+          '''
+          [profile foo]
+          aws_access_key_id = akid2
+          aws_secret_access_key = secret2
+          '''
+      helpers.mockHttpResponse 200, {}, '''
```

Comment:
I'm not sure this test is actually ensuring that the creds from `profile foo` are used instead of `default`.

What do you think about spying on `AWS.STS` or `AWS.Credentials` to get the accessKeyId that was used as the source?